From 1d1b4c2819d75653775db5df37a46a3d7f56de59 Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Tue, 6 Mar 2018 11:08:49 +0200
Subject: [PATCH 3429/5242] MLK-17671-1: drm: imx: dcss: add a delay after
 changing the pixel clock

commit  dfa64eede075b2c17dd48b3e17d78ea2194ba361 from
https://source.codeaurora.org/external/imx/linux-imx.git

DCSS needs some time to stabilize after switching to a new pixel clock.
All interrupts will delayed till the clock stabilizes and we'll end up
getting warnings about VBLANK interrupt taking more than 50ms to arrive.

This patch adds a 500ms delay after switching to a new clock. This will
allow DCSS to stabilize before enabling CRTC and DTG channels.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dcss/dcss-dtg.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/gpu/imx/dcss/dcss-dtg.c b/drivers/gpu/imx/dcss/dcss-dtg.c
index 9bab8ff..c939256 100644
--- a/drivers/gpu/imx/dcss/dcss-dtg.c
+++ b/drivers/gpu/imx/dcss/dcss-dtg.c
@@ -16,6 +16,7 @@
 #include <linux/bitops.h>
 #include <linux/io.h>
 #include <linux/clk.h>
+#include <linux/delay.h>
 #include <drm/drm_fourcc.h>
 
 #include <video/imx-dcss.h>
@@ -240,6 +241,8 @@ void dcss_dtg_sync_set(struct dcss_soc *dcss, struct videomode *vm)
 	clk_prepare_enable(dcss->pdiv_clk);
 	clk_prepare_enable(dcss->pout_clk);
 
+	msleep(500);
+
 	dcss_dtg_write(dtg, ((dtg_lrc_y << TC_Y_POS) | dtg_lrc_x),
 		       DCSS_DTG_TC_DTG);
 	dcss_dtg_write(dtg, ((dis_ulc_y << TC_Y_POS) | dis_ulc_x),
-- 
1.7.9.5

